src/main/java/com/users/service/UserService.java

```diff
@@ -0,0 +1,78 @@
+package com.users.service;
+
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Transactional
+@AllArgsConstructor
+@Service
+@Slf4j
+public class UserService {
+    private UserRepository userRepository;
+    private UserMapper userMapper;
+
+    public UserDto create(User user) {
+        log.info("User to be created {}", user);
+        List<User> allByEmail = userRepository.findAllByEmail(user.getEmail());
+        if (!allByEmail.isEmpty()) {
+            throw new EntityAlreadyExistsException(String.format("User with email %s already exists", user.getEmail()));
+        }
+        validateDateOfBirth(user.getDateOfBirth());
```

Comment:
in mod normal validarile se pun cat mai devreme in metoda ca sa nu fii nevoit sa faci un call in baza de date aiurea, dar request ul sa fie oricum invalid
tu ai deja o validare la nivel de dto, dar nu e rau sa mai validezi o data, dar pentru ca ai deja aceeasi metoda in validator, cauta o metoda astfel incat sa nu DRY

Reply:
banuiesc ca inainte de if trebuia, dar nu inteleg ce mi ai zis sa fac, pastrez metoda, fac alta sau raman doar cu validarea de pe Dto?

Reply:
inainte sa chemi findAllByEmail, ca daca validarea pica sa nu mai faci callul aiurea in baza de date. sa cauti o alternativa astfel incat sa nu ai metoda scrisa de 2 ori, si aici in service si in validatorul adult, sa o extragi undeva (intr-o clasa utilitara)